### PR TITLE
Rework the fetch-algorithms.ts file

### LINKED ISSUE #60: Fix bad data
There still are a few of errors in the data fetched from the repositories that need to be fixed.

- [x] [`tests`](https://the-algorithms.com/category/tests) get picked up
- [x] Categories [`strings`](https://the-algorithms.com/category/strings) and [`string`](https://the-algorithms.com/category/string) are almost the same
- [x] The category [`src`](https://the-algorithms.com/category/src) exists somehow
- [x] These are all the same: [`sorting`](https://the-algorithms.com/category/sorting) [`sorts`](https://the-algorithms.com/category/sorts) [`sorters`](https://the-algorithms.com/category/sorters) [`sort`](https://the-algorithms.com/category/sort)
- [x] These are also all the same: [`searching`](https://the-algorithms.com/category/searching) [`searches`](https://the-algorithms.com/category/searches) [`search`](https://the-algorithms.com/category/search)
- [x] Haskell [`specs`](https://the-algorithms.com/category/specs) get picked up
- [x] [`others`](https://the-algorithms.com/category/others) [`other`](https://the-algorithms.com/category/other) [`misc`](https://the-algorithms.com/category/misc) are the same
- [x] [`math`](https://the-algorithms.com/category/math) and [`maths`](https://the-algorithms.com/category/maths) are the same
- [x] [`hashing`](https://the-algorithms.com/category/hashing), [`hashes`](https://the-algorithms.com/category/hashes) and [`hash`](https://the-algorithms.com/category/hash) are the same
- [x] [`graph`](https://the-algorithms.com/category/graph) and [`graphs`](https://the-algorithms.com/category/graphs) are the same
- [x] [`divideconquer`](https://the-algorithms.com/category/divideconquer) and [`divideandconquer`](https://the-algorithms.com/category/divideandconquer) are the same
- [x] [`algorithmstests`](https://the-algorithms.com/category/algorithmstests) shouldn't exist
- [x] [`algorithms`](https://the-algorithms.com/category/algorithms) shouldn't be a category, instead the next tags of the algorithm

Some of them need to be fixed in the workflow that creates the `DIRECTORY.md` file directly in the seperate repositories, some can be fixed by expanding the [aliases](https://github.com/TheAlgorithms/website/blob/main/lib/aliases.js) to also include categories.

### PR DESCRIPTION
**Pull request by maltejur**
Rework the `fetch-algorithms.ts` file so that now all the repositories get cloned and the algorithms get pulled directly from there. This gets rid of the `DIRECTORY.md` files in the repositories, at least for the website. This new approach should also improve the build times (3 min local and 10 min vercel).

~~Because it has a very different structure, the C# repo isn't included yet, but apart from that, the final website should be pretty much the same as the current one.~~

Should fix everything in #60

Also opens #27 again, since that could be done now

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ node_modules
 .vercel
 .vscode
 cache
-out
\ No newline at end of file
+out
+/tmp
```
Comment (Panquesito7) [inline .eslintignore]: Missing an endline.
```suggestion
/tmp

```

File: .gitignore [hunk 2]
```diff
@@ -34,4 +34,5 @@ yarn-error.log*
 # vercel
 .vercel
 
-/cache
\ No newline at end of file
+/cache
+/tmp
```
Comment (Panquesito7) [inline .gitignore]:
```suggestion
/tmp

```

File: .vercelignore [hunk 3]
```diff
@@ -30,4 +30,5 @@ yarn-error.log*
 .env.test.local
 .env.production.local
 
-/cache
\ No newline at end of file
+/cache
+/tmp
```
Comment (Panquesito7) [inline .vercelignore]:
```suggestion
/tmp

```

**Review by Panquesito7 — COMMENTED** [review]
I am not very familiar with some of this code (and there are lots of files to review), so it would take a bit longer to review it all. Maybe I can just ensure everything works fine. 🙂

**Comment by maltejur** [comment]
Just testing and looking if everything still works is also much appreciated

**Review by Panquesito7 — DISMISSED** [review]
Everything seems to be working fine. LGTM. Thank you, @maltejur! 👍 🎉

**Comment by maltejur** [comment]
I still have to add C# somehow tho

**Review by Panquesito7 — DISMISSED** [review]

**Pull request merged**